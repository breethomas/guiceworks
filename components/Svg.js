// @flow
import React from 'react'

type SvgProps = {
  children?: React.Element<*>,
  width?: number,
  height?: number,
  size?: number,
}

const Svg = ({ children, size, width, height, ...props }: SvgProps) => (
  <svg
    aria-hidden
    height={size != null ? size : height}
    width={size != null ? size : width}
    {...props}
  >
    {children}
  </svg>
)

Svg.defaultProps = {
  children: null,
  width: 32,
  height: 32,
  size: null,
}

export default Svg
