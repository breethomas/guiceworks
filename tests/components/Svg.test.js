import React from 'react'
import renderer from 'react-test-renderer'
import Svg from '../../components/Svg'

describe('Svg', () => {
  it('shows the correct snapshot for the Svg component tree (width/height)', () => {
    const component = (
      <Svg width={32} height={32} viewBox="0 0 32 32">
        <g fill="currentColor">
          <polygon points="15.9 0 13.725 2.163 26.11 14.472 0 14.472 0 17.529 26.11 17.529 13.725 29.839 15.9 32 32 16" />
        </g>
      </Svg>
    )
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('shows the correct snapshot for the Svg component tree (size)', () => {
    const component = (
      <Svg size={32} viewBox="0 0 32 32">
        <g fill="currentColor">
          <polygon points="15.9 0 13.725 2.163 26.11 14.472 0 14.472 0 17.529 26.11 17.529 13.725 29.839 15.9 32 32 16" />
        </g>
      </Svg>
    )
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
