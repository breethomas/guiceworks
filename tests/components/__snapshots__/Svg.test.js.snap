// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Svg shows the correct snapshot for the Svg component tree (size) 1`] = `
<svg
  aria-hidden={true}
  height={32}
  viewBox="0 0 32 32"
  width={32}
>
  <g
    fill="currentColor"
  >
    <polygon
      points="15.9 0 13.725 2.163 26.11 14.472 0 14.472 0 17.529 26.11 17.529 13.725 29.839 15.9 32 32 16"
    />
  </g>
</svg>
`;

exports[`Svg shows the correct snapshot for the Svg component tree (width/height) 1`] = `
<svg
  aria-hidden={true}
  height={32}
  viewBox="0 0 32 32"
  width={32}
>
  <g
    fill="currentColor"
  >
    <polygon
      points="15.9 0 13.725 2.163 26.11 14.472 0 14.472 0 17.529 26.11 17.529 13.725 29.839 15.9 32 32 16"
    />
  </g>
</svg>
`;
